src/utils/build_url.ts

```diff
@@ -2,20 +2,18 @@ import type { ApiQueryParameters, Url } from '../types/options.js'
 
 const isBrowser = typeof location !== 'undefined' && typeof document !== 'undefined'
 
-const stringifyQuery = queryObj => new URLSearchParams(queryObj).toString()
-
-export function buildUrlFactory (instanceApiEndpoint: Url) {
+export function buildUrlFactory (instanceApiEndpoint: Url): BuildUrlFunction {
   return function (queryObj: ApiQueryParameters): Url {
     // Request CORS headers if the request is made from a browser
     // See https://www.wikidata.org/w/api.php ('origin' parameter)
     if (isBrowser) queryObj.origin = '*'
 
-    // Remove null or undefined parameters
-    Object.keys(queryObj).forEach(key => {
-      if (queryObj[key] == null) delete queryObj[key]
-    })
-
-    return instanceApiEndpoint + '?' + stringifyQuery(queryObj)
+    const queryArray = Object.entries(queryObj)
+      // Remove null or undefined parameters
+      .filter(([ _key, value ]) => typeof value !== 'undefined' && value !== null)
+      .map(([ key, value ]) => [ key, String(value) ])
```

Comment:
The stringification will already be done by `URLSearchParams#toString`, no?
```suggestion
```

Reply:
According to the types the constructor accepts `string | string[][] | Record<string, string> | URLSearchParams` which is why I added that line.